### DISCUSSION TITLE
想知道代码块折叠位置的文字是怎么写的呢

### DISCUSSION THREAD

**Discussion by gukaifeng** [ppoffice/hexo-theme-icarus#964] [Q&A]
这里是什么 markdown 扩展语法吗？  
我不知道怎么样写出一样的效果。  
这里不好发图，比如代码块上面折叠位置写的 `shell`，或者是 `_config.yml` 这种。

**Comment by ppoffice** [comment]
https://ppoffice.github.io/hexo-theme-icarus/Configuration/icarus%E7%94%A8%E6%88%B7%E6%8C%87%E5%8D%97-%E4%B8%BB%E9%A2%98%E9%85%8D%E7%BD%AE/#%E4%BB%A3%E7%A0%81%E9%AB%98%E4%BA%AE

**Reply by gukaifeng** [reply]
非常感谢，这个问题解决了！但是我还有另外的问题，这个写法已经不是markdown语法了，编辑的时候也无法预览出来，是否有更简单的，或者符合markdown语法的写法呢？如果没有的话，可不可以设置，上面那个文字默认显示语言，比如 `shell`，`Python` 这样，因为没有和这个的话，上面折叠的位置显得很空。

**Reply by gukaifeng** [reply]
我指的是，对于使用普通markdown语法的代码块（```语言 这个写法），能不能默认在其折叠的位置写上语言呢？

**Reply by eldicn** [reply]
@gukaifeng 使用下面的语法可以实现折叠

```
<details>
<summary>Coroutines</summary>

<br />

任何 MD 语法
</details>
```

<details>
<summary>Coroutines</summary>

<br />

```kotlin
suspend fun main() = coroutineScope {
    for (i in 0 until 10) {
        launch {
            delay(1000L - i * 10)
            print("$i ")
        }
    }
}
```
</details>

**Reply by lulu010722** [reply]
这需求太真实了，我也想要。

{% codeblock "可选文件名" lang:代码语言 >folded %}
...代码块内容...
{% endcodeblock %}

这种写法没有markdown引擎认识·。。希望找点更优雅的解决方案~

**Comment by lulu010722** [comment]
忽然发现hexo官网好像有直接使用```的解决方案了
https://hexo.io/docs/tag-plugins#Backtick-Code-Block
使用示例：

\`\`\`json abc.json
{
  "a": "1",
  "b": "2"
}
\`\`\`
